File: test/fixtures/small-dart/shape.dart
/// A 2-D shape.
abstract class Shape {
  double area();
}

/// Shared description behavior.
mixin Describable {
  String describe() => 'shape';
}

/// A circle with a radius.
class Circle extends Shape with Describable {
  final double radius;

  Circle(this.radius);

  @override
  double area() => radius * radius * 3;

  String describe() => 'circle';
}

/// The kinds of shape we render.
enum Kind { round, pointed }

/// Builds a label for a circle (extension method merged into Circle).
extension CircleLabel on Circle {
  String label() => describe();
}

Circle defaultCircle() => Circle(1.0);

File: test/fixtures/small-kotlin/shape.kt
package fixtures

/** A 2-D shape. */
interface Shape {
    fun area(): Double
}

/** A circle with a radius. */
data class Circle(val radius: Double) : Shape {
    override fun area(): Double {
        return radius * radius * 3
    }

    fun describe(): String {
        return "circle"
    }
}

/** The kinds of shape we render. */
enum class Kind {
    ROUND,
    POINTED
}

/** Builds a label for a circle (extension function). */
fun Circle.label(): String {
    return describe()
}

fun defaultCircle(): Circle {
    return Circle(1.0)
}
